dolphin/drivers/dell_emc/vmax/alert_handler.py

```diff
@@ -0,0 +1,159 @@
+
+    def __init__(self):
+        pass
+
+    """
+    Alert model contains below fields
+    category : Type of the reported notification
+    occur_time : Time of occurrence of alert. When trap does not contain it,
+                 it will be filled with receive time
+    match_key : This info uniquely identifies the fault point. Several infos
+                such as source system id, location, alarm id together can be
+                used to construct this
+    me_dn : Unique id at resource module (management system) side
+    me_name : Unique name at resource module (management system) side
+    native_me_dn : Unique id of the device at source system that reports the
+                   alarm
+    location : Alarm location information. It helps to locate the lowest unit
+               where fault is originated(Name-value pairs)
+               ex: Location = subtrack, No = 1, Slot No = 5.
+                   shelf Id = 1, board type = ADSL
+    event_type : Basic classification of the alarm. Probable values such as
+                 status, configuration, topology ....
+    alarm_id : Identification of alarm
+    alarm_name : Name of the alarm, might need translation from alarm id
+    severity : Severity of alarm. Helps admin to decide on action to be taken
+               Probable values: Critical, Major, Minor, Warning, Info
+    device_alert_sn : Sequence number of alert generated. This will be helpful
+                      during alert clearing process
+    manufacturer : Vendor of the device
+    Product_name : Name of the product
+    probable_cause : Probable reason for alert generation
+    clear_type : Alarm clearance type such as manual, automatic, reset clear
+    me_category : Resource category of the device generating the alarm
+                  Probable value: Network,Server,Storage..
+    """
+
+    def parse_alert(self, context, alert):
+        """Parse alert data got from alert manager and fill the alert model."""
+
+        # Check for mandatory alert attributes
+        for attr in self.mandatory_alert_attributes:
+            if not alert.get(attr):
+                msg = "Mandatory information %s missing in alert message. " \
+                      % attr
+                raise exception.ValidationError(detail=msg)
+
+        alert_model = {}
+        # These information are sourced from device registration info
+        alert_model['me_dn'] = alert['storage_id']
+        alert_model['me_name'] = alert['storage_name']
+        alert_model['manufacturer'] = alert['vendor']
+        alert_model['product_name'] = alert['model']
+
+        # trap info do not contain occur time, update with received time
+        # Get date and time. Format will be like : Wed May 20 01:53:29 2020
+        curr_time = datetime.now()
+        alert_model['occur_time'] = curr_time.strftime('%c')
+
+        # Fill all the alert model fields
+        if alert.get('category'):
```

Comment:
Instead of multiple if{} else{}, can we optimize like that:
add all default values at one plcae and update if other keys are present.
This will lead to good code readibility